packages/components/src/modal/index.tsx

```diff
@@ -75,7 +105,11 @@ function UnforwardedModal(
 	const headingId = title
 		? `components-modal-header-${ instanceId }`
 		: aria.labelledby;
-	const focusOnMountRef = useFocusOnMount( focusOnMount );
+
+	const focusOnMountRef = useFocusOnMount(
+		focusOnMount === 'firstElement' ? getFirstTabbableElement : focusOnMount
+	);
```

Comment:
Note with this change consumers of `Modal` will have to opt _into_ this behaviour by passing `firstElement`. The default of `Modal` is `true`.